dowhy/gcm/feature.py

```diff
@@ -0,0 +1,251 @@
+from dowhy.gcm.graph import get_ordered_predecessors, is_root_node, validate_node
+from dowhy.gcm.shapley import estimate_shapley_values, ShapleyConfig
+from dowhy.gcm.stats import marginal_expectation
+from dowhy.gcm.util.general import shape_into_2d, variance_of_matching_values, variance_of_deviations
+
+
+def feature_relevance(causal_model: StructuralCausalModel,
+                      target_node: Any,
+                      parent_samples: Optional[pd.DataFrame] = None,
+                      feature_attribution_func:
+                      Optional[Callable[[np.ndarray, np.ndarray], Union[np.ndarray, float]]] = None,
+                      num_background_samples: int = 5000,
+                      num_evaluation_samples: int = 500,
+                      max_batch_size: int = 100,
+                      shapley_config: Optional[ShapleyConfig] = None) -> Tuple[Dict[Any, Any], np.ndarray]:
+    """ Estimates the distribution based feature relevance of the inputs of the given target_node. This is, the 
+    relevance of direct parents with respect to the underlying causal model of target_node. Here, the unobserved noise 
+    is considered as a direct parent (input) as well. Samples utilized for the estimation are drawn from the given 
+    causal graph.
+    
+    By default, the used feature_attribution_func is based on the variance between Y and Y', where Y is the outputs of
+    the causal model and Y' the outputs of the models when certain features are randomized. In case of continuous data,
+    the feature relevance adds up to Var(Y - Y').
+    
+    Note: The feature relevance based on the distribution can not be directly compared with the feature relevance for 
+    single samples. If this is desired, the set function needs to be defined accordingly.
+    
+    Related paper:
+    Janzing, D., Minorics, L., & Bloebaum, P. (2020). 
+    Feature relevance quantification in explainable AI: A causal problem. 
+    In International Conference on Artificial Intelligence and Statistics (pp. 2907-2916). PMLR.
+    
+    :param causal_model: The fitted structural causal model.
+    :param target_node: Node with the causal model of interest.
+    :param parent_samples: Samples for the parents of the given target_node. If None is given, new samples are
+                           generated based on the graph.
+    :param feature_attribution_func: Set function for estimating the attributions of the model outcomes. This function
+                                     expects two inputs; the outcome of the causal model for some samples if certain features are permuted and the 
+                                     outcome of the model for the same samples when no features were permuted. The set functions represents the 
+                                     comparison between the samples, for instance, the variance of deviations.
+    :param num_background_samples: Number of samples used as background samples. Consider increasing this 
```

Comment:
good to rename background_samples as in previous PR

Reply:
We actually renamed it ```num_background_samples``` in the previous PR and clarified the meaning further in the docstring. I will try to clarify this here as well in the docstring. Might be a good idea to have a dedicated guide in the documentation to explain the Shapley estimator in general.

Reply:
sounds good! I like the idea of a dedicated guide for the shapley estimator, since it appears in so many functions.